### ISSUE TITLE
build fails if build path contains @

### ISSUE THREAD

**Issue by petere** [openssl/openssl#12078]
In 3.0.0-alpha3 and up to today's master HEAD (c4683009ad6e8d64e80112fd689921f6c169bd20), the configure phase of the build fails if the build path (i.e., the current directory) contains a `@` in the name. For example:

```
$ cd ~/tmp/foo@bar/openssl
$ ./config
Operating system: x86_64-apple-darwinDarwin Kernel Version 18.7.0: Mon Apr 27 20:09:39 PDT 2020; root:xnu-4903.278.35~1/RELEASE_X86_64
WARNING! If you wish to build 32-bit libraries, then you have to
         invoke 'KERNEL_BITS=32 ./config '.
Configuring OpenSSL version 3.0.0-alpha4-dev for target darwin64-x86_64-cc
Using os-specific seed configuration
Creating configdata.pm
Running configdata.pm
Possible unintended interpolation of @bar in string at configdata.pm line 28952.
Possible unintended interpolation of @bar in string at configdata.pm line 28953.
Global symbol "@bar" requires explicit package name (did you forget to declare "my @bar"?) at configdata.pm line 28952.
Global symbol "@bar" requires explicit package name (did you forget to declare "my @bar"?) at configdata.pm line 28953.
Execution of configdata.pm aborted due to compilation errors.
```

This is relevant because in Homebrew, package names may include a `@` in the name to designate a version, and this will affect the build path.  The existing `openssl@1.1` package does not appear to be affected.

**Comment by Croydon** [comment]
I guess I'm having the same problem while trying to create an OpenSSL package for Conan for the 3.x.x version series

```
[...]
openssl/3.0.0-alpha6: ['"conan-Debug-Macos-x86_64-apple-clang-11.0"', 'no-shared', '--prefix="/Users/jenkins/w/cci_PR-2054@2/.conan/data/openssl/3.0.0-alpha6/_/_/package/fd8adcac644fdf8ce8740c956c7bd7efe1f76b5d"', '--openssldir="/Users/jenkins/w/cci_PR-2054@2/.conan/data/openssl/3.0.0-alpha6/_/_/package/fd8adcac644fdf8ce8740c956c7bd7efe1f76b5d/res"', 'no-unit-test', 'threads', 'PERL=perl', 'no-tests', '--debug', '-fPIC', 'zlib-dynamic', '--with-zlib-include="/Users/jenkins/w/cci_PR-2054@2/.conan/data/zlib/1.2.11/_/_/package/0aa7be2d0914b6f3128105e305c90b9cb0cae86d/include"', '--with-zlib-lib="/Users/jenkins/w/cci_PR-2054@2/.conan/data/zlib/1.2.11/_/_/package/0aa7be2d0914b6f3128105e305c90b9cb0cae86d/lib"', 'zlib']

----Running------
> perl ./Configure "conan-Debug-Macos-x86_64-apple-clang-11.0" no-shared --prefix="/Users/jenkins/w/cci_PR-2054@2/.conan/data/openssl/3.0.0-alpha6/_/_/package/fd8adcac644fdf8ce8740c956c7bd7efe1f76b5d" --openssldir="/Users/jenkins/w/cci_PR-2054@2/.conan/data/openssl/3.0.0-alpha6/_/_/package/fd8adcac644fdf8ce8740c956c7bd7efe1f76b5d/res" no-unit-test threads PERL=perl no-tests --debug -fPIC zlib-dynamic --with-zlib-include="/Users/jenkins/w/cci_PR-2054@2/.conan/data/zlib/1.2.11/_/_/package/0aa7be2d0914b6f3128105e305c90b9cb0cae86d/include" --with-zlib-lib="/Users/jenkins/w/cci_PR-2054@2/.conan/data/zlib/1.2.11/_/_/package/0aa7be2d0914b6f3128105e305c90b9cb0cae86d/lib" zlib
-----------------
Configuring OpenSSL version 3.0.0-alpha6 for target conan-Debug-Macos-x86_64-apple-clang-11.0
Using os-specific seed configuration
Creating configdata.pm
Running configdata.pm
Possible unintended interpolation of @2 in string at configdata.pm line 21988.
Creating Makefile

**********************************************************************
***                                                                ***
***   OpenSSL has been successfully configured                     ***
***                                                                ***
***   If you encounter a problem while building, please open an    ***
***   issue on GitHub <https://github.com/openssl/openssl/issues>  ***
***   and include the output from the following command:           ***
***                                                                ***
***       perl configdata.pm --dump                                ***
***                                                                ***
***   (If you are new to OpenSSL, you might want to consult the    ***
***   'Troubleshooting' section in the INSTALL.md file first)      ***
***                                                                ***
**********************************************************************

----Running------
> /usr/bin/make -j8
-----------------
Makefile:2: *** missing separator.  Stop.
openssl/3.0.0-alpha6: 

JSON file created at '/Users/jenkins/w/cci_PR-2054@2/14/2524e93d-bb30-4d27-93e2-e5679c9d97dd/_create.json'
```

See also https://github.com/conan-io/conan-center-index/pull/2054

See e.g. https://github.com/conan-io/conan-center-index/pull/2054#issuecomment-699584747 for full logs for various configurations

**Comment by Croydon** [comment]
With `3.0.0-alpha7` I don't have this problem any longer. It would still be nice if someone could reference the PR/commit fixing this, so it is clear what caused it etc.

**Comment by Croydon** [comment]
Update: nevermind. It works now locally for me on Windows, but macOS CI (at least) still fails with the same error

https://github.com/conan-io/conan-center-index/pull/2054#issuecomment-709407010

**Comment by levitte** [comment]
Please try the fix in #13225
